test/test_runner_battle.c

```diff
@@ -1151,35 +1156,31 @@ static s32 TryMessage(s32 i, s32 n, const u8 *string)
 
         event = &DATA.queuedEvents[i].as.message;
         // Test_MgbaPrintf("Looking for: %S Found: %S\n", event->pattern, string); // Useful for debugging.
-        for (j = k = 0; ; j++, k++)
+        j = 0; k = 0;
+        while (1)
         {
-            if (event->pattern[k] == CHAR_SPACE)
-            {
-                switch (string[j])
-                {
-                case CHAR_SPACE:
-                case CHAR_PROMPT_SCROLL:
-                case CHAR_PROMPT_CLEAR:
-                case CHAR_NEWLINE:
-                    j++;
-                    k++;
-                    break;
-                }
+            if (IS_SKIPOVER_CHAR(event->pattern[k])) {
+                k++;
+                continue;
             }
-            if (event->pattern[k] == EOS)
-            {
-                // Consume any trailing '\p'.
-                if (string[j] == CHAR_PROMPT_CLEAR)
-                    j++;
+            if (IS_SKIPOVER_CHAR(string[j])) {
+                j++;
+                continue;
+            }
```

Comment:
Doesn't this essentially ignore all whitespace inside the pattern and output? e.g. `"Hello World"` would match `"HelloWorld"` and vice-versa?

The intent of the original code was that a space in the pattern would match any whitespace in the actual output, so I think this change should say something like "try to match 1 or more skipover characters in `event->pattern,` and if so match 1 or more skipover characters in `string`"?

Example incorrectly-passing test (no engine code modified):
```diff
WILD_BATTLE_TEST("Pokemon gain exp after catching a Pokemon")
{
    u8 level = 0;

    PARAMETRIZE { level = 50; }
    PARAMETRIZE { level = MAX_LEVEL; }

    GIVEN {
        PLAYER(SPECIES_WOBBUFFET) { Level(level); }
        OPPONENT(SPECIES_CATERPIE) { HP(1); }
    } WHEN {
        TURN { USE_ITEM(player, ITEM_ULTRA_BALL); }
    } SCENE {
-       MESSAGE("You used Ultra Ball!");
+       MESSAGE("YouusedUltraBall!");
        ANIMATION(ANIM_TYPE_SPECIAL, B_ANIM_BALL_THROW, player);
        if (level != MAX_LEVEL) {
            EXPERIENCE_BAR(player);
        }
    }
}
```
Example incorrectly passing code (no tests modified):
```diff
-static const u8 sText_PlayerUsedItem[] = _("You used\n{B_LAST_ITEM}!");
+static const u8 sText_PlayerUsedItem[] = _("Youused{B_LAST_ITEM}!");
```